pkcs11/_pkcs11.pyx

```diff
@@ -1550,6 +1550,19 @@ cdef class lib:
         except StopIteration:
             return token
 
+    def wait_for_slot_event(self, blocking=False):
+        cdef CK_FLAGS flag = 0
+        cdef CK_SLOT_ID slot_id
+
+        if not blocking:
+            flag = 1
+
+        rv = _funclist.C_WaitForSlotEvent(flag, & slot_id, NULL)
+        assertRV(rv)
+        if rv == CKR_NO_EVENT:
+            return -1
+        return slot_id
```

Comment:
Let's return the actual `Slot` object?

Reply:
I like the idea of only returning the slot_id and giving the developer the option of scanning for the slot or not. In some cases it won't make sense e.g. If the event is for removal/detachment, there won't be a Slot to return.

Reply:
How would you know if the `slot_id` is a slot or not? It makes sense to me to do that work for them and return a slot.

Reply:
How would you feel about returning all available slots?

Reply:
I don't know what we'd do with all slots. Is there a reason we can't return the slot with that id?

Reply:
No I guess not, I went ahead an added that.